src/utils.rs

```diff
@@ -156,42 +158,42 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
 
 /// Derive a deterministic pseudorandom value in `[-n, n]` from the
 /// [`Transcript`].
-pub(crate) fn plusminus_bn_random_from_transcript(
+pub(crate) fn plusminus_challenge_from_transcript(
     transcript: &mut Transcript,
-    n: &BigNumber,
-) -> BigNumber {
+) -> Result<BigNumber> {
     let mut is_neg_byte = vec![0u8; 1];
     transcript.challenge_bytes(b"sampling negation bit", is_neg_byte.as_mut_slice());
     let is_neg: bool = is_neg_byte[0] & 1 == 1;
 
     // The sampling method samples from the open interval, so add 1 to sample from
     // the _closed_ interval we want here.
-    let open_interval_max = n + 1;
-    let b = positive_bn_random_from_transcript(transcript, &open_interval_max);
-    match is_neg {
+    let q = k256_order();
+    let open_interval_max = &q + 1;
+    let b = positive_challenge_from_transcript(transcript, &open_interval_max)?;
+    Ok(match is_neg {
         true => -b,
         false => b,
-    }
+    })
 }
 
-/// Derive a deterministic pseduorandom value in `[0, n)` from the
-/// [`Transcript`].
-pub(crate) fn positive_bn_random_from_transcript(
+pub(crate) fn positive_challenge_from_transcript(
```

Comment:
todo: add the documentation back in. The open vs closed boundaries are kind of nitpicky and I prefer to have explicit docs about them.